.github/workflows/Format.yml

```
@@ -0,0 +1,42 @@
+name: Format Check
+on:
+  # because we rely on ReviewDog to trigger failure, there's no reason to run on anything
+  pull_request:
+    paths:
+      - "**/*.jl"
+      - ".github/workflows/FormatCheck.yml"
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: only if it is a pull request build.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
```

I've been using this, FWIW
```suggestion
name: Format Check
on:
  push:
    branches:
      - 'main'
      - /^release-.*$/
    tags: ['*']
  pull_request:
    types: [opened, synchronize, reopened, ready_for_review]
concurrency:
  # Skip intermediate builds: always.
  # Cancel intermediate builds: only if it is a pull request build.
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}

```

---

right, see the comment ;) 

build won't fail unless reviewdog actually runs, and that only runs on PRs (non-draft in this workflow)